MaterialLib/MPL/MaterialSpatialDistributionMap.h

```diff
@@ -0,0 +1,43 @@
+/**
+ * \date   Nov 28, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+namespace MeshLib
+{
+template <typename PROP_VAL_TYPE>
+class PropertyVector;
+}  // namespace MeshLib
+
+namespace MaterialPropertyLib
+{
+class Medium;
+
+class MaterialSpatialDistributionMap
+{
+private:
+    std::map<int, std::unique_ptr<Medium>> const& _media;
```

Comment:
As far as I understand the domain of the map is [0, #number of media]. Maybe a vector can be used here?

Reply:
Note to myself; compare with error/warning message code.

Reply:
@TomFischer the media ids follow numbering from MaterialIDs and can be rather arbitrary, especially negative. We might have a look on it later, if it looks like a bottleneck.